.eslintrc

```diff
@@ -1,19 +1,37 @@
 {
+  "env": {
+    "browser": true,
+    "es6": true,
+    "jest": true
+  },
+  "globals": {
+    "Atomics": "readonly",
+    "SharedArrayBuffer": "readonly"
+  },
   "root": true,
   "parser": "@typescript-eslint/parser",
   "parserOptions": {
     "ecmaVersion": 6,
     "sourceType": "module",
     "ecmaFeatures": {
       "jsx": true
-    }
+    },
+    "project": "./tsconfig.json"
   },
-  "plugins": ["@typescript-eslint"],
+  "plugins": ["react", "@typescript-eslint", "jest"],
   "extends": [
-    "react-app",
+    "airbnb-typescript",
+    "airbnb/hooks",
+    "plugin:@typescript-eslint/recommended",
+    "plugin:jest/recommended",
+    "plugin:prettier/recommended",
     "plugin:react/recommended",
     "plugin:@typescript-eslint/eslint-recommended",
-    "plugin:react-hooks/recommended"
+    "plugin:react-hooks/recommended",
+    "prettier",
+    "prettier/react",
+    "prettier/@typescript-eslint",
+    "react-app"
   ],
```

Comment:
These sound good as a starting point, not deeply familiar with them, but let's do it!

- https://github.com/prettier/eslint-plugin-prettier#recommended-configuration
  - > This plugin works best if you disable all other ESLint rules relating to code formatting, and only enable rules that detect patterns in the AST. (If another active ESLint rule disagrees with prettier about how code should be formatted, it will be impossible to avoid lint errors.) You can use `eslint-config-prettier` to disable all formatting-related ESLint rules.
  - > If your desired formatting does not match the prettier output, you should use a different tool such as `prettier-eslint` instead.

Are we using `eslint-config-prettier` here (as recommended in the above)?

- https://nicedoc.io/prettier/eslint-config-prettier

Reply:
https://prettier.io/docs/en/integrating-with-linters.html mentions:

- `eslint-config-prettier`
- `tslint-config-prettier`
- `stylelint-config-prettier`

The notes section specifically talks about best practices, what to use, what is less optimal to use, etc:

- https://prettier.io/docs/en/integrating-with-linters.html#notes

Reply:
I'm not 100% certain the difference between `plugins` & `extends`, but we may as well get as strict as reasonably possible, and insist that we gradually move toward zero lint issues..